### DISCUSSION TITLE
How do I enable features in project

### DISCUSSION THREAD

**Discussion by uzytkownik** [emacs-rustic/rustic#32] [Q&A]
This is something I tried to google but despite trying several thing I could not figure out how. How do I enable feature flags in rustic?

**Comment by psibi** [comment]
What do you mean by that? Do you want to pass feature flag to the build command?

**Reply by uzytkownik** [reply]
I meant for the rust-analyzer.

**Reply by psibi** [reply]
Probably you are looking for `lsp-rust-analyzer-cargo-extra-args`: https://emacs-lsp.github.io/lsp-mode/page/lsp-rust-analyzer/#lsp-rust-analyzer-cargo-extra-args ?

**Reply by uzytkownik** [reply]
@psibi Tried that. Didn't worked. I probably did something wrong but I had hard time figuring out why.

**Reply by psibi** [reply]
Maybe restart the language server if you haven't already so that it can take the new variables ?

**Reply by uzytkownik** [reply]
@psibi I did tried that. I displayed the variables to check if it's set.

**Reply by timmc** [reply]
If you're having the same problem I did (Emacs not doing full checking of feature-gated code on save) then you probably want `(lsp-rust-analyzer-cargo-watch-args ["--all-features"])` (watch-args, not extra-args).

**Reply by timmc** [reply]
I wonder if this is something that could be made default in rustic. I don't really understand how all the different Emacs packages involved here interact, though...

**Comment by CeleritasCelery** [comment]
For example if you want to run `cargo build` with a particular feature you would do `C-u C-c C-c C-b` and then type the command line you want (e.g. `--features feat1,feat2` or `--all-features`). If you set `rustic-cargo-use-last-stored-arguments` to `t` it will remember the flags you set next time you call a command.

**Reply by uzytkownik** [reply]
I meant for the rust-analyzer.

**Reply by CeleritasCelery** [reply]
All rust analyzer features are bound to variables that start with `lsp-rust-analyzer-*`. (e.g. `lsp-rust-analyzer-diagnostics-enable`, `lsp-rust-analyzer-import-group`, etc).
